### ISSUE TITLE
Question regarding Implementation details.

### ISSUE THREAD

**Issue by fabrahman** [qkaren/Counterfactual-StoryRW#10]
Hi Karen,

I am trying to modify and implement your code on my own dataset. I encountered a few issues regarding that I appreciate your guidance.
1- I realized even in your dataset ``` len(x1 + ' ' + x2 + ' ' + y + ' | ' + x1 + ' ' + xx2 + ' ' + y + ' ') > max_seq_len``` most of the time and thus often times, the instance (y or yy) is truncated in [here](https://github.com/qkaren/Counterfactual-StoryRW/blob/d16f8fb5ac7b829ad39d6c2afec4ab83dd1c0975/utils/data_utils_rewriting.py#L75), doesn't it cause to loose information especially in y and yy which we want to predict?

2- I decided to have max_seq_len of 200 to be sufficient enough to cover my instances, however even with batch_size 2 it throw Out Of Memory error. Did you used batch_size of 2 ? how long does it take for you to train your model on a single gpu? what was your gpu specification?

3- With batch size of 1 it is of course too slow. Is there anyway I can make it faster. I do have two gpu 11G if the code can be run by multiple GPUs.

4- In the paper you mentioned you ran it for 10k iterations and did an early stop. Does the code here support early stop? And do you remember on what epoch it stopped training? 

5- And finally loading the 345M gpt-2 takes a long time, did you tried caching it?

Many tanks for your responses.

**Comment by qkaren** [comment]
Truncation does lose information I think. Using a larger max_seq_len could be better, depending on your GPU memory size, etc.

I used 1080Ti GPUs. The training did not take a long time, but I didn't remember clearly how long exactly. The training loop periodically does evaluation on dev set and saves the best model checkpoint. If you observe the dev loss is going up after a best model checkpoint for a while, I guess it's safe to stop the training.

The newer Texar version supports faster GPT2 loading.  For this codebase, a trick is to load the pre-trained GPT-2, save a checkpoint immediately, and then use `--checkpoint` ([here](https://github.com/qkaren/Counterfactual-StoryRW/blob/d16f8fb5ac7b829ad39d6c2afec4ab83dd1c0975/rewriting_main.py#L712)) to initialize the model in the subsequent runs

**Comment by Hannabrahman** [comment]
@qkaren Many thanks. So you just used the ```FLAGS.distributed``` for running on multiple GPUs, no other thing to be taken care of (especially regarding the loss)?

**Comment by qkaren** [comment]
Yes, `FLAGS.distributed` is only for distributed running on multi GPUs
